### ISSUE TITLE
SEO core hardening: 8 transforms/renderer fixes para que futuros imports lleguen indexable-ready

### ISSUE THREAD

**Issue by AlfonsoSBLA** [Growth4U-systems/alarife-payload#44]
## Contexto

Auditoría SEO de Paymático (2026-05-21) detectó 8 problemas que NO son específicos del cliente: son gaps del core de Alarife que cada nuevo import volverá a sufrir. Esta issue agrupa los fixes que viven en `apps/cms/src/lib/import/transforms.ts`, `apps/web/src/layouts/BaseLayout.astro`, `apps/web/src/pages/sitemap.xml.ts` y `apps/web/src/lib/sanitize-html.ts`.

Cada sub-tarea es idempotente, opt-in donde aplica vía `transforms` list, y se beneficia inmediatamente Paymático + cualquier site futuro (HC, Monzo, etc.).

Evidencia detallada del audit: `/Users/alfonsosb/PROJECTS/Growth Raistlin/memory/2026-05-21.md` (sesión SEO audit).

---

## B1 · `deriveImageAlts` transform

**Síntoma**: 80/92 `<img alt="">` en home Paymático.

**Causa**: pipeline de import migra Media pero deja `Media.alt` vacío (campo required que se bypassa con overrideAccess) y NO toca `<img alt="">` dentro del rawHtml.

**Fix**: nuevo PASS en `transforms.ts`:

* Recorre `pages[].sections[0].html`, busca `<img>` con `alt=""` o sin attribute
* Mapeo en este orden:
  1. `Brand.imageAltDictionary` (nuevo array `[{filename, alt}]` editable en admin)
  2. Si Media tiene `alt` no-vacío → usar ese (sincroniza con Media collection)
  3. Derivado del filename: kebab/snake → "Title Case", strip extensión, strip `-p-NNN` Webflow suffix, strip `-N` Payload dedup suffix
* También rellena `Media.alt` para items con alt vacío usando la misma cascada (cumple required del campo y unifica fuente de verdad)
* Reporta: `{imagesUpdated, altsFromDictionary, altsFromMedia, altsFromFilename, altsLeftEmpty}`

**Tests**: extender `apps/cms/src/lib/import/__tests__/transforms.test.ts` con fixture de Webflow export con alts vacíos.

---

## B2 · Heading-spacing transform

**Síntoma**: H2 home Paymático "Una plataforma.Todos tus canales." (sin espacio tras punto), "negocioscon" (concatenación).

**Causa**: Webflow exporta `<br>` que linkedom interpreta como string-join sin separador en algunos casos.

**Fix**: añadir al PASS existente (o nuevo `normalizeHeadings`):

* Aplicar SOLO a `<h1>..<h6>` content:
  * `/\.([A-ZÁÉÍÓÚÑ])/g` → `. $1`
  * `/([a-záéíóúñ])([A-ZÁÉÍÓÚÑ])/g` → `$1 $2` (catch concatenaciones tipo "negocioscon")
* Idempotente (no daña texto ya correcto)
* No tocar `<style>` ni `<script>` ni contenido fuera de headings

**Tests**: fixture con strings concretos del export Webflow Paymático.

---

## B3 · Extract `Brand.chromeCss` to a `<link>` stylesheet

**Síntoma**: 437 KB de `<style>` inline por página, replicado 52 veces. Sin caché HTTP entre navegaciones.

**Causa**: `BaseLayout.astro:234` emite `{chromeCss && <style set:html={chromeCss} />}` directamente.

**Fix**:

1. Crear endpoint Astro `apps/web/src/pages/chrome.css.ts` que fetcha `brand.chromeCss` y responde con `Content-Type: text/css; charset=utf-8` + `Cache-Control: public, max-age=31536000, immutable` + `ETag` (sha256 del contenido)
2. En `BaseLayout.astro`: cambiar el `<style set:html={chromeCss}>` por `<link rel="stylesheet" href="/chrome.css">` cuando hay `chromeCss`
3. `prepareEmbeddedHtml` ya soporta dejar fuera el chrome CSS (porque pages no lo reemiten — el `headExtras` page-specific es lo que sigue inline, está bien)

**Beneficio medible**: -400 KB/página en navegaciones cacheadas. LCP móvil debería caer >1s en 4G.

---

## B4 · Strip legacy JSON-LD + `<SchemaJsonLd />` Astro component

**Síntoma**: 4 bloques `<script type="application/ld+json">` en home Paymático con Organization+WebSite redundantes.

**Causa**: Webflow tenía bakeado el schema en el body. `rawHtml.html` lo trae literal. Layout no emite schema propio porque no quiere duplicar.

**Fix**:

1. Nuevo transform `stripLegacySchema` (idempotente):
   * Elimina `<script type="application/ld+json">` de `section.html`
   * Hace dump a `Site.legacySchemaSnapshot` (JSON read-only) para auditoría/recuperación
2. Nuevo componente `apps/web/src/components/SchemaJsonLd.astro`:
   * Input: `pageType`, `page` (path/title/description/excerpt/author/publishedAt), `brand` (organization data), `site` (website data), `faqEntries`
   * Output: **un solo** `<script type="application/ld+json">` **con** `@graph` y `@id` cruzados:
     * `Organization` (@id `${siteUrl}#organization`)
     * `WebSite` (@id `${siteUrl}#website`, publisher → org)
     * `WebPage`/`Article`/`FAQPage` según pageType
   * Para `blog-post`: usa Article con author relación
   * Para `legal`: omite schema (página noindex)
3. Brand global recibe campos opcionales: `regulatorIdentifier`, `socialProfiles[]` (array), `contactPoint` (objeto)
4. `BaseLayout.astro` invoca `<SchemaJsonLd ... />` después del head principal

**Tests**: snapshot test del JSON-LD generado para los 5 pageTypes principales.

---

## B5 · Title formula: normalizar comparación accent-insensitive

**Síntoma**: home Paymático `<title>` = "Paymático: plataforma de pagos regulada | Paymatico" (brand duplicado con/sin tilde).

**Causa**: `BaseLayout.astro:73`: `const fullTitle = title.includes(siteName) ? title : \`${title} | ${siteName}\``. `"Paymático: ...".includes("Paymatico")\` → false → añade.

**Fix**:<br>\`\`\`ts<br>const norm = (s: string) => s.normalize('NFD').replace(/\[̀-ͯ\]/g, '').toLowerCase()<br>const fullTitle = norm(title).includes(norm(siteName)) ? title : `${title} | ${siteName}`<br>\`\`\`

5 líneas. Cualquier brand con acentos o casing distinto deja de duplicarse.

---

## B6 · Per-tier OG image fallback

**Síntoma**: todas las páginas comparten `og-default.png`.

**Causa**: `BaseLayout.astro:75` único fallback `defaultOgImageUrl`.

**Fix**:

1. Brand global gana `defaultOgImageByPageType: group { home, landing, blogIndex, blogPost, legal, pseo, funnel, partner }` (cada uno upload→media)
2. BaseLayout cascada:<br>`Page.meta.ogImage` → `Brand.defaultOgImageByPageType[pageType]` → `Brand.defaultOgImage` → `${siteDomain}/og-default.png`
3. `[...path].astro` ya pasa `pageType` al layout (verificar; si no, añadirlo a props)

**Tests**: para cada pageType, snapshot del `og:image` rendered.

---

## B7 · Sitemap `lastmod` granular

**Síntoma**: 52 URLs con mismo `lastmod` (fecha del build).

**Fix**: en `apps/web/src/pages/sitemap.xml.ts`, usar `page.updatedAt` (Payload lo expone) como `<lastmod>`. \~3 líneas.

---

## B8 · Extender `sanitize-html.ts` rewrite a imágenes + url(...) en CSS

**Síntoma**: 28 referencias hardcoded a `www.paymatico.com` + 2 a Webflow CDN + 1 a googleapis jQuery en home Paymático tras el fix de hoy.

**Causa**: `rewriteRelativeUrls` solo toca `<a href>`, no `<img src>`, `<source srcset>`, `<link href>`, ni `url(...)` dentro de `<style>`.

**Fix**:

1. Extender el regex pass en `sanitize-html.ts` para cubrir: `src=`, `srcset=` (parsear cada candidato), `url(...)` en CSS embebido
2. Sobre las hardcoded externas a Webflow CDN (`d3e54v103j8qbb.cloudfront.net`, `ajax.googleapis.com`): detección en `runRewriteAssetLeaks` (transforms.ts), opción de strip o warning en import report
3. Configurable: si el editor quiere mantener jQuery o un CDN, lo añade a `Site.allowedExternalAssets[]`

**Tests**: fixture con un body Webflow que mezcla `<a>`, `<img>`, `<style>url()`, y `<link href="cdn">`.

---

## Definición de "Done"

- [ ] Cada sub-tarea con su PR (o agrupadas si comparten archivo). Sugerido: PR1 B1+B2 (transforms.ts), PR2 B3+B5+B6 (BaseLayout+brand), PR3 B4 (schema), PR4 B7+B8 (sitemap+sanitize)
- [ ] Tests en `apps/cms/src/lib/import/__tests__/` para todos los transforms
- [ ] Re-importar Paymático con los nuevos transforms y verificar:
  * 0 imágenes con alt vacío
  * H2 con espaciado correcto
  * 1 solo bloque `<script type="application/ld+json">`
  * `<link rel="stylesheet" href="/chrome.css">` en lugar de `<style>` inline grande
  * `<title>` sin brand duplicado
  * OG distinto por tipo de página
  * Sitemap con `lastmod` real por URL
  * 0 referencias hardcoded a `www.paymatico.com` en HTML rendered

## Bloqueo / dependencias

Ninguna. Cada bloque es independiente. Issue trackea el conjunto pero los PRs son paralelizables.

## Auditoría de origen

Generada por audit SEO en sesión Claude Code 2026-05-21 sobre `localhost:4321/` (Paymático Astro dev). Reportes:

* 437 KB CSS inline / página
* 80/92 imágenes alt vacío
* 4 JSON-LD redundantes
* 28 refs hardcoded `www.paymatico.com`
* Canonical apunta a `paymatico.local` (no env-driven)
* Title brand duplicado
* Single OG image todo el site
* Sitemap lastmod único
